### ISSUE TITLE
Three similar shaped DIVs break the tests when using mouse clicking

### ISSUE THREAD

**Issue by djungowski** [casperjs/casperjs#999]
Consider the following the (simplified) document:

``` html
<!DOCTYPE html>
<head></head>
<body>
<div id="foobar"></div>
<div class="something-before"></div>
<div class="wrapper">
    <div class="more-wrapping">
        <div class="foobar user-action">first link</div>
        <div class="foobar">first ipsum</div>
    </div>

    <div class="more-wrapping">
        <div class="foobar user-action">second link</div>
        <div class="foobar">second ipsum</div>
    </div>

    <div class="more-wrapping">
        <div class="foobar user-action">third link</div>
        <div class="foobar">third ipsum</div>
    </div>
</div>
<div class="something-after"></div>

<script type="text/javascript">
    function updateFoobar(content) {
        document.getElementById('foobar').innerHTML = content;
    }

    var elements = document.getElementsByClassName('foobar user-action');
    for(var i = 0; i < elements.length; i++) {
        console.log(elements[i]);
        elements[i].addEventListener('click', function(event) {
            updateFoobar(event.target.innerHTML);
        });
    }
</script>
</body>
```

with the following test code:

``` javascript
casper.test.begin('Testcase', function(test) {
    casper.start('./testcase.html', function() {
        this.click('.more-wrapping:nth-child(1) .foobar');
        test.assertSelectorHasText('#foobar', 'first link');
    });

    casper.thenOpen('./testcase.html', function() {
        this.click('.more-wrapping:nth-child(2) .foobar');
        test.assertSelectorHasText('#foobar', 'second link');
    });

    casper.thenOpen('./testcase.html', function() {
        this.click('.more-wrapping:nth-child(3) .foobar');
        test.assertSelectorHasText('#foobar', 'third link');
    });

    casper.run(function() {
        test.done();
    });
});
```

This will throw the following error:

```
FAIL Find "third link" within the selector "#foobar"
#    type: assertSelectorHasText
#    file: testcase.js:15
#    code: test.assertSelectorHasText('#foobar', 'third link');
#    subject: false
#    selector: "#foobar"
#    text: "third link"
#    actualContent: ""
```

Ways to make it work:
1. Changing "nth-child(3)" to "nth-of-type(3)"
2. Changing ".more-wrapping:nth-child(3) .foobar" to ".more-wrapping:nth-child(3) .user-action"
3. Removing `<div class="foobar">second ipsum</div>`
4. Changing the class name of `<div class="foobar">second ipsum</div>`
5. Using :last-child

Methods 1 and 2 for some reason work in this case but do **not** work with my actual case. Methods 3 and 4 are not an option / in my case not possible. Method 5 is possible in this example but not necessarily in all examples.

**Comment by BIGjuevos** [comment]
**As part of a cleanup effort:** 
Looks to be a stale help request. Assuming you've moved on to better and cooler things.  Please feel free to re-open if this is still an active issue for you and we'll try our best to help out.

**Comment by djungowski** [comment]
Yes, it's stale because no one has replied to it ever. Wtf.

**Comment by istr** [comment]
I can verify this issue for PhantomJS 1.9.7. It is fixed with PhantomJS 2.1.1.

See https://github.com/istr/casperjs/tree/999-mouseclick

**Comment by istr** [comment]
@BIGjuevos @mickaelandrieu Let us decide about setting the "wontfix" flag for this and add a documentation hint (i.e. ```:nth-child()``` is broken in PhantomJS 1.9.* in some situations).

**Comment by BIGjuevos** [comment]
@istr What would a fix for this look like within CasperJS?

**Comment by BIGjuevos** [comment]
@djungowski 
> Yes, it's stale because no one has replied to it ever. Wtf.

Thanks for pinging back and letting us know.  Many of these issues were not originally triaged.  I'm trying to help out by cleaning things to get a better picture of what is actual work and what was help requests.  We're now looking into the issue.  Thanks for your patience.

**Comment by istr** [comment]
> What would a fix for this look like within CasperJS?

Very unsure as this seems to be a WebKit bug, basically. These are CSS3 selectors and the support in older WebKits was flaky. This is why I requested to discuss "wontfix" and add the viable solutions @djungowski provided to the documentation.

**Comment by kensoh** [comment]
@djungowski confirming that the issue is resolved with PhantomJS v2.1.1. I'm running on CasperJS v1.1.4. Below is the output for your test script/html. Closing the issue for now, but do ping if you still run into issue with this :) I'm a CasperJS user for couple of years. Browsing through open issues now to see if anything I can help with.

```
# Testcase
PASS Find "first link" within the selector "#foobar"
PASS Find "second link" within the selector "#foobar"
PASS Find "third link" within the selector "#foobar"
PASS Testcase (NaN test)
PASS 3 tests executed in 0.204s, 3 passed, 0 failed, 0 dubious, 0 skipped.      
```
